### PR TITLE
(GH-250) Added support to upload Cake.Issues report to appveyor

### LINKED ISSUE #250: Add Cake.Issues.PullRequests.AppVeyor
We could add the Cake.Issues.PullRequests.AppVeyor addin which would report found analyzer issues to AppVeyor as shown [here](https://cake-contrib.github.io/Cake.Issues.Website/docs/pull-request-systems/appveyor/examples/write-messages).

Together with a GitHub Pull Request Notification they can also be written as a comment to pull requests as shown [here](https://cake-contrib.github.io/Cake.Issues.Website/docs/pull-request-systems/appveyor/examples/github-pullrequest-integration).

This would be a temporary solution until a better implementation with GitHub checks is available (See https://github.com/cake-contrib/addin-requests/issues/3).

Drawback is that issue would be written in individual requests to AppVeyor which can take some time if there are a few issues.

### PR DESCRIPTION
**Pull request by AdmiringWorm**
This PR adds support for uploading issues to appveyor that was created through `Cake.Issues`.

Test run:
Task: https://ci.appveyor.com/project/AdmiringWorm/cake-recipe/builds/24316246#L4456
Results: https://ci.appveyor.com/project/AdmiringWorm/cake-recipe/builds/24316246/messages

fixes #250 

/CC @pascalberger @gep13

### CODE REVIEW

**Comment by AdmiringWorm** [comment]
> One of the unsolved issues with Cake.Issues.Recipe is to load addins conditionally. > If you're interested in investigating in this I would be happy to accept a PR

I actually tried to look into that here, before submitting the PR.
Tried using the preprocessor and environment variable to conditionally install the addins (only got compile errors), also tried restoring them through RequireAddin and run a different script (cake was unable to reference the addins in this case).

I did try a couple of other things as well (forgot what those were now though), without success.

So basically, I don't think Cake is at the right point just yet that would allow us conditional addins :(

**Comment by gep13** [comment]
This has been rebased and merged here: https://github.com/cake-contrib/Cake.Recipe/commit/0dd7aaf7dfa0bacbaf72a2d26435ff34fb914890

**Comment by gep13** [comment]
@AdmiringWorm your changes have been merged, thanks for your contribution 👍
